src/main/java/formflow/library/config/submission/Action.java

```diff
@@ -25,4 +28,23 @@ public default void run(Submission submission) {
   public default void run(Submission submission, String id) {
     throw new UnsupportedOperationException("Not implemented");
   }
+
+  /**
+   * Runs an action on form submission data to potentially manipulate the data.
+   *
+   * @param formSubmission form submission object the action is associated with, not null
+   */
+  public default void run(FormSubmission formSubmission) {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  /**
+   * Runs an action on form submission data to potentially manipulate the data.
+   *
+   * @param formSubmission form submission object the action is associated with, not null
+   * @param id             id for the iteration
+   */
+  public default void run(FormSubmission formSubmission, String id) {
```

Comment:
The more I think about this, the more I realize that the FormSubmission actions probably don't need the id part (for uuid) for running an action. But I could be wrong here. Maybe it's better to be flexible and provide this option.  Thoughts?

Reply:
Yeah - Flexibility would nice but I dont think we'll need it for the encryption workaround so I'll leave it out for now